src/Symfony/Bundle/FrameworkBundle/Routing/DelegatingLoader.php

```diff
@@ -85,7 +85,9 @@ public function load($resource, $type = null)
         $this->loading = false;
 
         foreach ($collection->all() as $route) {
-            if ($controller = $route->getDefault('_controller')) {
+            $controller = $route->getDefault('_controller');
+            // If $controller is in class::method notation already skip it for performance.
+            if ($controller && false === strpos($controller, '::')) {
```

Comment:
What do you think about using a guard clause here to reduce the complexity of the code?

```php
if (!$controller || false !== strpos($controller, '::')) {
    continue;
}
```